lib/helpers.js

```diff
@@ -103,6 +104,26 @@ exports.parseAndValidateAddressString = function(address, defaultPort) {
     });
 };
 
+async function readDir(dir, fileCallback = () => {}) {
```

Comment:
good point. no this wouldn't makes sense as a promisified function - the point is to call the callback for each file, instead of allocating memory for a giant array of files. I'll remove the default, check the type, and early out if it's not a function.